echopop/inversion/inversion_matrix.py

```diff
@@ -0,0 +1,1230 @@
+        minimizer objects for each measurement location.
+
+        Parameters
+        ----------
+        model_parameters : InvParameters
+            Container with biological and physical parameters including:
+            - Organism characteristics (length_mean, g, h, etc.)
+            - Parameter bounds and vary flags for optimization
+            - Current parameter values as initial estimates
+        model_settings : Dict[str, Any]
+            Model configuration dictionary containing:
+            - type: Scattering model identifier (e.g., "pcdwba")
+            - Model-specific settings (taper_order, frequency_interval, etc.)
+            - Distribution parameters for length and orientation averaging
+            - Environment parameters (sound_speed_sw, density_sw)
+
+        Raises
+        ------
+        EchopopValidationError
+            If parameters or settings fail validation against model schema
+
+        Notes
+        -----
+        This method performs several important setup tasks:
+        1. Validates parameter compatibility with the specified model type
+        2. Scales parameters to [0,1] if scale_parameters=True in simulation_settings
+        3. Loads the appropriate scattering model function from the registry
+        4. Creates minimizer objects for each measurement (computationally intensive)
+
+        Parameter scaling is recommended for optimization performance as it
+        normalizes all parameters to similar ranges, improving numerical
+        conditioning of the optimization problem.
+
+        The model function is retrieved from SCATTERING_MODEL_PARAMETERS registry
+        based on the model type, ensuring the correct forward model is used.
+
+        Examples
+        --------
+        >>> inv_matrix = InversionMatrix(data, simulation_settings)
+        >>> parameters = InvParameters(param_dict)
+        >>> settings = {"type": "pcdwba", "taper_order": 10.0}
+        >>> inv_matrix.build_scattering_model(parameters, settings)
+        """
+
+        # Validate
+        try:
+            # ---- Check
+            valid_args = ValidateBuildModelArgs.create(
+                **dict(model_parameters=model_parameters, model_settings=model_settings)
+            )
+        # Break creation
+        except (ValidationError, Exception) as e:
+            raise EchopopValidationError(str(e)) from None
+
+        # Update attributes
+        self.model_params = valid_args["model_parameters"]
+        self.model_settings.update(valid_args["model_settings"])
+
+        # Update the model settings to include the model function
+        self.model_settings["model_function"] = SCATTERING_MODEL_PARAMETERS.get(
+            self.model_settings["type"]
+        )["function"]
+
+        # Scale the parameters, if defined
+        if self.simulation_settings["scale_parameters"] and not self.model_params.is_scaled:
+            self.model_params.scale()
```

Comment:
I think having the scaled=True/False both in `simulation_settings` and `model_params` and have one controls the other is messy. Suggest having this flag only in `model_params`, and as suggested in a previous comment, keep both the scaled and unscaled version of the parameters.